NeuroID/NeuroIDClass/Extensions/NIDSession.swift

```diff
@@ -130,62 +86,88 @@ public extension NeuroID {
-            saveEventToLocalDataStore(createNIDSessionEvent())
+        // The following events have to happen for either
+        //  an existing session that begins a new flow OR
+        //  a new session with a new flow
+        // 1. Determine if flow should be sampled
+        // 2. CREATE_SESSION and MOBILE_METADATA events captured
+        // 3. Capture ADV (based on global config and lib installed)
 
-            captureMobileMetadata()
+        // If SDK is already started, update config and continue
+        if NeuroID.isSDKStarted {
+            configService.updateConfigOptions(siteID: siteID) {
+                NeuroID.updateIsSampledStatus()
 
-            checkThenCaptureAdvancedDevice()
-        }
+                // capture CREATE_SESSION and METADATA events for new flow
+                saveEventToLocalDataStore(createNIDSessionEvent())
+                captureMobileMetadata()
 
-        if !startStatus.started {
-            return startStatus
-        }
+                checkThenCaptureAdvancedDevice()
 
-        // add linkedSite var
-        NeuroID.linkedSiteID = siteID
-
-        let setLinkedSiteIDEvent = NIDEvent(sessionEvent: NIDSessionEventName.setLinkedSite)
-        setLinkedSiteIDEvent.v = siteID
-        saveEventToLocalDataStore(setLinkedSiteIDEvent)
+                NeuroID.addLinkedSiteID(siteID)
+                completion(SessionStartResult(true, NeuroID.getUserID()))
+            }
+        } else {
+            // If the SDK is not started we have to start it first
+            //  (which will get the config using passed siteID)
 
-        return startStatus
+            // if userID passed then startSession should be used
+            if userID != nil {
+                _ = NeuroID.startSession(siteID: siteID, sessionID: userID) { startStatus in
+                    NeuroID.addLinkedSiteID(siteID)
+                    completion(startStatus)
+                }
+            } else {
+                _ = NeuroID.start(siteID: siteID) { started in
+                    NeuroID.addLinkedSiteID(siteID)
+                    completion(SessionStartResult(started, NeuroID.getUserID()))
+                }
+            }
+        }
     }
 }
 
 extension NeuroID {
-    static func validateSiteID(_ string: String) -> Bool {
-        let regex = #"^form_[a-zA-Z0-9]{5}\d{3}$"#
-        let predicate = NSPredicate(format: "SELF MATCHES %@", regex)
-
-        let valid = predicate.evaluate(with: string)
+    /*
+      Determine if the session/flow should be sampled (i.e. events captured and sent)
+       if not then change the _isSessionFlowSampled var
+       this var will be used in the DataStore.cleanAndStoreEvent method
+       and will drop events if false
+     */
+    static func updateIsSampledStatus() {
+        let maxSampleRate = 100
+        if NeuroID.configService.configCache.currentSampleRate >= maxSampleRate {
+            NeuroID._isSessionFlowSampled = true
```

Comment:
So if its above 100? we are going to sample it?

Reply:
Yes we are going to sample (sample = collect) if it's 100 or higher. Again this was to catch a weird edge case where if the value equaled the min/max it wouldn't behave as expected